src/pymoca/backends/casadi/model.py

```diff
@@ -524,16 +524,42 @@ def simplify(self, options):
                         value = None
 
                     if variable is not None:
-                        del alg_states[variable.name()]
-
-                        variables.append(variable)
                         if eq.is_op(ca.OP_SUB):
-                            values.append(value)
+                            return variable, value
                         else:
-                            values.append(-value)
+                            return variable, -value
 
-                        # Skip this equation
-                        continue
+                if eq.n_dep() == 2 \
+                    and eq.is_op(ca.OP_ADD) \
```

Comment:
Can this also be generalized to OP_SUB?

Reply:
The reason it is set up like this is that casadi if_else() produces two if_else_zero() ops and adds them up. I will add a comment.